packages/backend/src/api/1/handlers/auth.js

```diff
@@ -1,3 +1,33 @@
+import crypto from 'crypto';
 import express from 'express';
+import passport from 'passport';
 
-module.exports = express.Router();
+const router = express.Router();
+
+router.get('/reddit/callback', (req, res, next) => {
+  // Check for origin via state token
+  if (req.query.state === req.session.state)
+    passport.authenticate('reddit', {
+      successRedirect: '/auth_status',
+      failureRedirect: '/login_failure',
+    })(req, res, next);
+  else next(new Error(403));
+});
+
+router.get('/reddit', (req, res, next) => {
+  if (!req.session.state)
+    req.session.state = crypto.randomBytes(32).toString('hex');
+  passport.authenticate('reddit', {
+    state: req.session.state,
+    duration: 'permanent',
+  })(req, res, next);
+});
+
+router.get('/auth_status', (req, res) =>
+  res.json({
+    auth_status: req.isAuthenticated(),
```

Comment:
You can actually just check if user is null / not null, so I doubt auth_status field is really needed